src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts

```diff
@@ -0,0 +1,121 @@
+import {Hunk, Patch, RepoDomain} from '../../../types';
+import {logger} from '../../../logger';
+import {buildSummaryComment} from './message-handler';
+
+const COMFORT_PREVIEW_HEADER =
+  'application/vnd.github.comfort-fade-preview+json';
+
+/**
+ * Multiline comment GitHub API parameters
+ * For more information see
+ * https://developer.github.com/v3/pulls/comments/#create-a-review-comment-for-a-pull-request
+ */
+interface MultilineComment {
+  body: string;
+  path: string;
+  start_line: number;
+  line: number;
+  side: 'RIGHT' | 'LEFT';
+  start_side: 'RIGHT' | 'LEFT';
+}
+
+/**
+ * GitHub-defined type. The Octokit library/docs are probably behind since create review already
+ * accept multi-line code comments. However, the API does not reflect that.
+ */
+export type PullsCreateReviewParamsComments = {
+  path: string;
+  position: number;
+  body: string;
+};
+
+/**
+ * Convert the patch suggestions into GitHub parameter objects.
+ * Use this to generate review comments
+ * For information see:
+ * https://developer.github.com/v3/pulls/comments/#create-a-review-comment-for-a-pull-request
+ * @param suggestions
+ */
+export function buildReviewComments(
+  suggestions: Map<string, Patch[]>
+): MultilineComment[] {
+  const fileComments: MultilineComment[] = [];
+  suggestions.forEach((patches: Patch[], fileName: string) => {
+    patches.forEach(patch => {
+      const comment: MultilineComment = {
+        path: fileName,
+        body: `\`\`\`suggestion\n${patch.newContent}\`\`\``,
+        start_line: patch.start,
+        line: patch.end,
+        side: 'RIGHT',
+        start_side: 'RIGHT',
+      };
+      fileComments.push(comment);
+    });
+  });
+  return fileComments;
+}
+
+/**
+ * Make a request to GitHub to make review comments
+ * @param octokit
+ * @param suggestions
+ * @param remote
+ * @param pullNumber
+ */
+export async function makeInlineSuggestions(
+  octokit: Octokit,
+  suggestions: Map<string, Patch[]>,
+  outOfScopeSuggestions: Map<string, Hunk[]>,
+  remote: RepoDomain,
+  pullNumber: number
+): Promise<void> {
+  const comments = buildReviewComments(suggestions);
+  if (!comments.length) {
+    logger.info('No valid suggestions to make');
+  }
+  if (!comments.length && !outOfScopeSuggestions.size) {
+    logger.info('No suggestions were generated. Exiting...');
+    return;
+  }
+  const summaryComment = buildSummaryComment(outOfScopeSuggestions);
+  if (summaryComment) {
+    logger.warn('Some suggestions could not be made');
+  }
+  // apply the suggestions to the latest sha
+  // the latest Pull Request hunk range includes
+  // all previous commit valid hunk ranges
+  const headSha = (
+    await octokit.pulls.get({
+      owner: remote.owner,
+      repo: remote.repo,
+      pull_number: pullNumber,
+    })
+  ).data.head.sha;
+  await octokit.pulls.createReview({
+    owner: remote.owner,
+    repo: remote.repo,
+    pull_number: pullNumber,
+    commit_id: headSha,
+    event: 'COMMENT',
```

Comment:
Feature request: pipe an option to make this 'REQUEST_CHANGES' or 'COMMENT'. This way someone could configure it to be a blocking review.